documentation/api/wire_format/replace_catalog_inputs_format_v1.markdown

```diff
@@ -0,0 +1,74 @@
+---
+title: "PuppetDB: Replace catalog inputs wire format, version 1 (experimental)"
+layout: default
+canonical: "/puppetdb/latest/api/wire_format/replace_catalog_inputs_v1.html"
+---
+
+The v1 `replace catalog inputs` command tells PuppetDB to replace the current
+set of catalog inputs with the new.
+
+Changing the value of this setting for a certname has the same effect
+as receipt of a new catalog with respect to extending the lifetime of
+the node.
+
+Replace catalog inputs command format
+-----
+
+### Version
+
+This is **version 1** of the replace catalog inputs command.
+
+### Encoding
+
+The command is serialized as JSON, which requires strict UTF-8 encoding.
+
+### Main data type: Replace catalog inputs for node
+
+     {
+      "certname": <string>,
+      "producer_timestamp": <datetime>,
+      "catalog_uuid": <string>,
+      "inputs": [[<string>, <string>]],
+     }
+
+#### `certname`
+
+String. The name of the node for which the catalog was compiled for.
+
+#### `producer_timestamp`
+
+DateTime. The time of command submission from the Puppet master to PuppetDB,
+according to the clock on the Puppet master.
+
+`producer_timestamp` is optional but *highly* recommended. When provided, it is
+used to determine the precedence between this command and other commands that
+modify the same node. This field is provided by, and should thus reflect the
+clock of, the Puppet master.
+
+When `producer_timestamp` is not provided, the PuppetDB server's local time is
+used. If another command is received for a node while a non-timestamped
+"deactivate node" command is pending processing, the results are *undefined*.
+
+#### `catalog_uuid`
+
+The uuid of the catalog this input was used for. Since not every catalog will
+provide a record of its inputs, and the catalog is replaced via a different
+command, this may not be the same as the current catalog being stored in
+PuppetDB.
+
+#### `inputs`
+
+The `inputs` to a catalog are a list of tuples, where the first element is its
+`type` (ie. `"hiera"`) and the second is a unique name for the inputs (for a hiera
```

Comment:
"unique name for the input" (singular), or maybe even "...where the first element is the input type (e.g. "hiera"), and the second is a unique name (for a hiera key ...".